src/synapse/azext_synapse/custom.py

```diff
@@ -0,0 +1,171 @@
+    )
+    return client.create_or_update(resource_group_name, workspace_name, workspace_info)
+
+
+def update_workspace(cmd, client, resource_group_name, workspace_name, sql_admin_login_password=None,
+                     provisioning_state=None, identity_type="SystemAssigned", principal_id=None, tags=None):
+    identity = ManagedIdentity(type=identity_type)
+    workspace_patch_info = WorkspacePatchInfo(
+        tags=tags,
+        identity=identity,
+        sql_administrator_login_password=sql_admin_login_password,
+        provisioning_state=provisioning_state
+    )
+    return client.update(resource_group_name, workspace_name, workspace_patch_info)
+
+
+def create_spark_pool(cmd, client, resource_group_name, workspace_name, spark_pool_name, location,
+                      spark_version="2.4", node_size=NodeSize.medium.value, node_count=3,
+                      node_size_family=NodeSizeFamily.memory_optimized.value, auto_scale_enabled=True,
+                      min_node_count=3,
+                      max_node_count=40, auto_pause_enabled=True, delay_in_minutes=15, spark_events_folder="/events",
+                      library_requirements_filename=None, library_requirements_content=None,
+                      default_spark_log_folder="/logs", force=False):
+    big_data_pool_info = BigDataPoolResourceInfo(location=location, spark_version=spark_version, node_size=node_size,
+                                                 node_count=node_count, node_size_family=node_size_family,
+                                                 spark_events_folder=spark_events_folder,
+                                                 default_spark_log_folder=default_spark_log_folder)
+    if auto_scale_enabled:
+        big_data_pool_info.auto_scale = AutoScaleProperties(enabled=auto_scale_enabled, min_node_count=min_node_count,
+                                                            max_node_count=max_node_count)
+    if auto_pause_enabled:
+        big_data_pool_info.auto_pause = AutoPauseProperties(enabled=auto_pause_enabled,
+                                                            delay_in_minutes=delay_in_minutes)
+
+    if library_requirements_filename or library_requirements_content:
+        big_data_pool_info.library_requirements = LibraryRequirements(filename=library_requirements_filename,
+                                                                      content=library_requirements_content)
+    return client.create_or_update(resource_group_name, workspace_name, spark_pool_name, big_data_pool_info, force)
+
+
+def update_spark_pool(cmd, client, resource_group_name, workspace_name, spark_pool_name, tags=None):
+    return client.update(resource_group_name, workspace_name, spark_pool_name, tags)
+
+
+def create_sql_pool(cmd, client, resource_group_name, workspace_name, sql_pool_name, location, sku_tier=None,
+                    sku_name="DW1000c", max_size_bytes=None, source_database_id=None, recoverable_database_id=None,

```

Comment:
>"DW1000c" [](start = 29, length = 9)

Strongly suggest not to provide default SKU. This might be different in regions. Along with time being, this might also change.

Reply:
fixed

---
In reply to: [386418279](https://github.com/Azure/azure-cli-extensions/pull/1339#discussion_r386418279) [](ancestors = 386418279)